services/session.go

```diff
@@ -48,6 +54,49 @@ func (s *Session) AddNewClient(c *Client) {
 	s.clients = append(s.clients, c)
 }
 
+func (s *Session) SchedulePeriodicTasks() {
+	if s.scheduled {
+		return
+	}
+
+	go func() {
+		s.scheduled = true
+
+		s.ticker = time.NewTicker(1 * time.Second)
+		for range s.ticker.C {
+			var wg = sync.WaitGroup{}
+			wg.Add(len(s.Instances))
+			for _, i := range s.Instances {
+				if i.dockerClient == nil {
+					// Need to create client to the DinD docker daemon
+
+					transport := &http.Transport{
+						DialContext: (&net.Dialer{
+							Timeout:   1 * time.Second,
+							KeepAlive: 30 * time.Second,
+						}).DialContext}
+					cli := &http.Client{
+						Transport: transport,
+					}
+					c, err := client.NewClient(fmt.Sprintf("http://%s:2375", i.IP), client.DefaultVersion, cli, nil)
+					if err != nil {
+						log.Println("Could not connect to DinD docker daemon", err)
+					} else {
+						i.dockerClient = c
+					}
+				}
+				go func() {
+					for _, t := range periodicTasks {
+						t.Run(i)
+					}
+					wg.Done()
```

Comment:
Isn't better to defer wg.Done()? if a panic is raised then Done() won't be called.